Sources/Orbit/Support/Layout/ScreenLayoutModifier.swift

```diff
@@ -1,72 +1,72 @@
 import SwiftUI
 
-/// Padding to apply in a screen layout.
+/// Padding to apply in a `screenLayout` context.
 public enum ScreenLayoutPadding {
     /// Padding of `medium` size in compact horizontal size and `xxLarge` size in regular horizontal size.
     case `default`
     /// Padding of `medium` size in both compact and regular horizontal size.
     case compact
     /// Custom fixed padding.
     case custom(horizontal: CGFloat = .medium, top: CGFloat = .medium, bottom: CGFloat = .medium)
-}
-
-struct ScreenLayoutModifier: ViewModifier {
-
-    @Environment(\.horizontalSizeClass) var horizontalSizeClass
-
-    let edges: Edge.Set
-    let padding: ScreenLayoutPadding
-    let maxContentWidth: CGFloat
-
-    func body(content: Content) -> some View {
-        content
-            .environment(\.isInsideScreenLayout, true)
-            .padding(.top, edges.contains(.top) ? topPadding : 0)
-            .padding(.leading, edges.contains(.leading) ? horizontalPadding : 0)
-            .padding(.trailing, edges.contains(.trailing) ? horizontalPadding : 0)
-            .padding(.bottom, edges.contains(.bottom) ? bottomPadding : 0)
-            .frame(maxWidth: maxContentWidth)
-            .frame(maxWidth: .infinity)
-    }
 
-    var horizontalPadding: CGFloat {
-        switch padding {
-            case .default:                      return defaultPadding
+    public func horizontal(horizontalSizeClass: UserInterfaceSizeClass?) -> CGFloat {
+        switch self {
+            case .default:                      return `default`(horizontalSizeClass: horizontalSizeClass)
             case .compact:                      return .medium
             case .custom(let horizontal, _, _): return horizontal
         }
     }
 
-    var topPadding: CGFloat {
-        switch padding {
-            case .default:                      return defaultPadding
+    public func top(horizontalSizeClass: UserInterfaceSizeClass?) -> CGFloat {
+        switch self {
+            case .default:                      return `default`(horizontalSizeClass: horizontalSizeClass)
             case .compact:                      return .medium
             case .custom(_, let top, _):        return top
         }
     }
 
-    var bottomPadding: CGFloat {
-        switch padding {
-            case .default:                      return defaultPadding
+    public func bottom(horizontalSizeClass: UserInterfaceSizeClass?) -> CGFloat {
+        switch self {
+            case .default:                      return `default`(horizontalSizeClass: horizontalSizeClass)
             case .compact:                      return .medium
             case .custom(_, _, let bottom):     return bottom
         }
     }
 
-    var defaultPadding: CGFloat {
+    public func `default`(horizontalSizeClass: UserInterfaceSizeClass?) -> CGFloat {
         horizontalSizeClass == .regular
```

Comment:
Suggest to use switch-case.

Reply:
It would be unnecessary larger though, with optional size class.